### PR TITLE
Clean up Comparator lib, act two

### PR DESCRIPTION
**Pull request by chschlue**
Followup to #2467, forgot to add OC symbols as announced at #2450.

Example:
MCP6566

---

All contributions to the kicad library must follow the [KiCad library convention](http://kicad-pcb.org/libraries/klc/)

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [ ] Provide a URL to a datasheet for the symbol(s) you are contributing
- [ ] An example screenshot image is very helpful
- [ ] Ensure that the associated footprints match the [official footprint library](https://github.com/kicad/kicad-footprints)
  - A new fitting footprint must be submitted if the library does not yet contain one.
- [ ] If there are matching footprint PRs, provide link(s) as appropriate
- [ ] Check the output of the Travis automated check scripts - fix any errors as required
- [ ] Give a reason behind any intentional library convention rule violation.

### CODE REVIEW

**Comment by cpresser** [comment]
I finally found motivation to continue with this. Sorry for the huge delay.

This is a list of changed parts:
* MIC845N :heavy_check_mark: 
* MCP5969 :heavy_check_mark: 
* MCP6567 (Pin6 change) :heavy_check_mark: 
* MCP6566U :heavy_check_mark: 
* MCP6566 :heavy_check_mark: 
* ADCMP356 (some polylines moved) :heavy_check_mark: 
* ADCMP350 (some polylines moved) :heavy_check_mark: 

Those are missing any keyword/description that says `open`. I did verify that all of them actually are open-collector devices by looking at the datasheets.
* LMV393 (Pin6 change)
* LMV331
* LM339
* LM319
* LM311
* LM319H
* LM2903
* LM2901
* MCP65R46

When searching for `Comparator open` I would expect for all of them to show up. But there are way less shown.
image

How about also improving the description/keywords?

**Comment by chschlue** [comment]
Done.

File: Comparator.dcm [hunk 1]
```diff
@@ -2,73 +2,73 @@ EESchema-DOCLIB  Version 2.0
...
 D Single-input comparator, internal 0.6V reference, Active-Low Open-Drain output, KS-4
-K single comparator with reference
+K cmp collector
 F https://www.analog.com/media/en/technical-documentation/data-sheets/ADCMP350_354_356.pdf
 $ENDCMP
 #
 $CMP ADCMP354
 D Single-input comparator, internal 0.6V reference, Active-High Open-Drain output, KS-4
-K single comparator with reference
+K cmp collector
```
Comment (cpresser) [inline Comparator.dcm:17]: missing 'open' here?

File: Comparator.dcm [hunk 2]
```diff
@@ -2,73 +2,73 @@ EESchema-DOCLIB  Version 2.0
...
 D Ultrafast 7 ns Single Supply Comparator, DIP-8/SOIC-8/TSSOP-8
-K comparator ultrafast
+K cmp complementary push-pull
 F https://www.analog.com/media/en/technical-documentation/data-sheets/ad8561.pdf
 $ENDCMP
 #
 $CMP ADCMP350
 D Single-input comparator, internal 0.6V reference, Active-Low Open-Drain output, KS-4
-K single comparator with reference
+K cmp collector
```
Comment (cpresser) [inline Comparator.dcm:11]: missing 'open' here?

File: Comparator.dcm [hunk 3]
```diff
@@ -2,73 +2,73 @@ EESchema-DOCLIB  Version 2.0
...
 D Single-input comparator, internal 0.6V reference, Active-High Push-Pull output, KS-4
-K single comparator with reference
+K cmp
 F https://www.analog.com/media/en/technical-documentation/data-sheets/ADCMP350_354_356.pdf
 $ENDCMP
 #
 $CMP LM2901
 D Quad Differential Comparators, DIP-14/SOIC-14/TSSOP-14
-K quad comparator
+K cmp open collector
```
Comment (cpresser) [inline Comparator.dcm:29]: I think the quad keyword is usefull. Why did you remove it? There are similar occurrences below with 'dual'.

File: Comparator.dcm [hunk 4]
```diff
@@ -80,187 +80,187 @@ $ENDCMP
...
 D Micro-Power Comparator / Battery Monitor, Active-Low Push-Pull Output, SC-70-5
-K single comparator battery monitor
+K single cmp
 F http://ww1.microchip.com/downloads/en/DeviceDoc/mic845.pdf
 $ENDCMP
 #
 $CMP MIC845N
 D Micro-Power Comparator / Battery Monitor, Active-High Open-Drain Output, SC-70-5
-K single comparator battery monitor
+K single cmp collector
```
Comment (cpresser) [inline Comparator.dcm:245]: this has the single keyword, where other have the dual/quad keyword removed.

File: Comparator.dcm (hunk 1 above)
Comment (chschlue) [reply Comparator.dcm:17]: 'open' already in description

File: Comparator.dcm (hunk 2 above)
Comment (chschlue) [reply Comparator.dcm:11]: 'open' already in description

File: Comparator.dcm (hunk 3 above)
Comment (chschlue) [reply Comparator.dcm:29]: 'quad' already in description

File: Comparator.dcm (hunk 4 above)
Comment (chschlue) [reply Comparator.dcm:245]: 'single' not already in description

**Comment by cpresser** [comment]
Tbh, I did not look at the keywords descriptions in that way. My mind was focusing around the fact that it is somewhat inconsistent when looking at different parts.
image

That in mind, I think we can merge this PR as is. Further cleaning up Descriptions/Keywords can be done in a separate PR if someone finds the time&motivation for that.
What do you think?

**Comment by chschlue** [comment]
I generally used whatever the datasheet title said as the description and added keywords if anything I deemed useful was omitted (minus the stuff I missed, as above). Not sure if that's the best way, but I don't think we can "normalize" all comp descriptions without making them excessively long.
Also, I already have #2670 in the works, splitting non-fully-specified symbols, normalizing triangle styles and such.
